### PR TITLE
udpate mqtt client auth

### PR DESCRIPTION
**Pull request by suchen-sci**
This pr fix issue #505 by remove auth file and put auth usernames and passwords into yaml file. For security, we use salted SHA256 checksum to protect password.

### CODE REVIEW

File: pkg/filter/mqttclientauth/mqttauth.go [hunk 1]
```diff
@@ -48,20 +44,21 @@ type (
...
-	// Spec is spec for MQTTClientAuth
-	// authFile format:
-	// - username: test
-	//   passBase64: dGVzdA==
+	// Spec is spec for MQTTClientAuth.
+	// For security of password, passwords in yaml file should be salted SHA256 checksum.
+	// password = sha256sum(connect.password + salt)
 	Spec struct {
-		AuthFile string `yaml:"authFile" jsonschema:"required"`
+		Salt string  `yaml:"salt" jsonschema:"omitempty"`
```
Comment (localvar) [inline pkg/filter/mqttclientauth/mqttauth.go:54]: is `Salt` a fixed value for all user/password pairs? then it is helpless for security and can be remvoed.

Comment (suchen-sci) [reply pkg/filter/mqttclientauth/mqttauth.go:54]: it is fixed. I think it mainly used for solving rainbow table attack. making such table may need some time and user can update during that time. One salt for one password is also ok...

Comment (zhao-kun) [reply pkg/filter/mqttclientauth/mqttauth.go:54]: I had a similar question about this, but I think @suchen-sci's explanation makes sense.

**Review by zhao-kun — APPROVED** [review]

File: pkg/filter/mqttclientauth/mqttauth.go (hunk 1 above)
Comment (localvar) [reply pkg/filter/mqttclientauth/mqttauth.go:54]: Yes, got it, it could be some help though I think building a rainbow table won't take too much time.

**Review by localvar — APPROVED** [review]

File: pkg/filter/mqttclientauth/mqttauth.go (hunk 1 above)
Comment (suchen-sci) [reply pkg/filter/mqttclientauth/mqttauth.go:54]: so one salt for one password? and also support salt for all passwords, let user to choose. If they want more secure use one salt for one passwrod?

**Pull request merged**